src/renderer/dx/CustomTextRenderer.cpp

```diff
@@ -220,6 +222,144 @@ using namespace Microsoft::Console::Render;
                               clientDrawingEffect);
 }
 
+// Function Description:
+// - Attempt to draw the cursor. If the cursor isn't visible or on, this
+//   function will do nothing. If the cursor isn't within the bounds of the
+//   current run of text, then this function will do nothing.
+// - This function will get called twice during a run, once before the text is
+//   drawn (underneath the text), and again after the text is drawn (above the
+//   text). Depending on if the cursor wants to be drawn above or below the
+//   text, this function will do nothing for the first/second pass
+//   (respectively).
+// Arguments:
+// - d2dContext - Pointer to the current D2D drawing context
+// - textRunBounds - The bounds of the current run of text.
+// - drawingContext - Pointer to structure of information required to draw
+// - firstPass - true if we're being called before the text is drawn, false afterwards.
+// Return Value:
+// - S_FALSE if we did nothing, S_OK if we successfully painted, otherwise an appropriate HRESULT
+[[nodiscard]] HRESULT _drawCursorHelper(::Microsoft::WRL::ComPtr<ID2D1DeviceContext> d2dContext,
+                                        D2D1_RECT_F textRunBounds,
+                                        const DrawingContext& drawingContext,
+                                        const bool firstPass)
+try
+{
+    if (!drawingContext.cursorInfo.has_value())
+    {
+        return S_FALSE;
+    }
+
+    const auto& options = drawingContext.cursorInfo.value();
+
+    // if the cursor is off, do nothing - it should not be visible.
+    if (!options.isOn)
+    {
+        return S_FALSE;
+    }
+
+    // TODO GH#6338: Add support for `"cursorTextColor": null` for letting the
+    // cursor draw on top again.
+    if (!firstPass)
+    {
+        return S_FALSE;
+    }

```

Comment:
Oh is that your reason for having both passes call it? Okay. Could also do what Dustin said in the other comment and scope it to just block cursors for now...

Reply:
Also, I think this sets you up in good position to reintroduce the "invert" cursor, perhaps on color=null, because you can easily see the FG and BG color of the text about to be printed right here and invert those brushes.

Reply:
Honestly, #6338 is for someone who _wants_ `filledBox` _on top of the character_. I don't know who this person is, but you know how back-compat be